lib/iotagent-mqtt.js

```diff
@@ -144,6 +145,24 @@ function mqttMessageHandler(topic, message) {
 }
 
 /**
+ * Generate the list of topics related to the device, based on the device attribute definitions.
+ *
+ * @param {Object} device           Device object containing all the information about the provisioned device.
+ * @return {Array}                  List of strings names of the topics associated to the selected device.
+ */
+function generateDeviceTopics(device) {
+    var topics = [];
+
+    for (var i = 0; i < device.active.length; i++) {
```

Comment:
If it exists it will always be an Array (as it is returned from the same IOTA library, it would be a pretty bad bug otherwise). I'm also 98% sure it will always exist, as it comes from the Mongoose object storing the device, but I'll add that check, anyways.